### ISSUE TITLE
How to resume training at certain epoch or checkpoint?

### ISSUE THREAD

**Issue by itsjustrafli** [ultralytics/yolov5#11684]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Long story short: So I was training my yolov5s model on my google account that is not subscribed to colab pro. Apparently, I need more compute units, so I get myself a subscription. But ... I was mistakenly subscribed colab pro on the other google account (dumb me), so I exported the trained yolov5 directory and uploaded to the other google account that has colab pro subs. After all moving process are done, I did the resume training using this command:

`!python train.py --resume`

The training resumed at 33 epochs, but after 42 epochs, I realized that cache for my training data is load only for 1843 train images (actual train data is 13k+ images). So I want to get restart training at 33 epochs.

I wonder, how to start training at the 33 epochs?

Before posting this question, I have read about resume training. But it explained for resume at epoch where it gets interrupted, not at defined checkpoint (certain epoch).

TL;DR: How to start training at defined check point?

Thank you in advance responding to my issue!

**Comment by glenn-jocher** [comment]
@itsjustrafli to start training at a specific checkpoint, you can use the `--resume` option followed by the path to the checkpoint file you want to start from. For example, to start training from the checkpoint at 33 epochs, you can use the following command:

`!python train.py --resume /path/to/checkpoints/epoch_33.pt`

Please note that the specified checkpoint file should be valid and match the architecture of your model. Let us know if you have any further questions or concerns.

**Comment by itsjustrafli** [comment]
Can you tell me where is the checkpoints path is located? I didn't find any in the yolov5s directory

**Comment by glenn-jocher** [comment]
@itsjustrafli hello! By default, YOLOv5 checkpoints are saved in the `runs/train` directory in your YOLOv5 repository. The checkpoints are saved as PyTorch `.pt` files with names like `epoch_10.pt`, where `epoch_10` represents the epoch number of the checkpoint. If you don't see the `runs/train` directory in your repository, you may need to train your model first using the `train.py` script. Let us know if you have any further questions or concerns!

**Comment by itsjustrafli** [comment]
Ah, found it! Thank you so much @glenn-jocher!

**Comment by glenn-jocher** [comment]
You're welcome, @itsjustrafli! I'm glad I could help you find the answer you were looking for. If you have any further questions or concerns, please feel free to let us know. We're always here to help.

**Comment by itsjustrafli** [comment]
Wait, I got another question on resuming training.

If I want to resuming training, did I have to add all defined arguments that I used before? So not just only this command:

`!python train.py --resume`

But all the arguments I defined before + `--resume` arguments, like ...

`!python --device 0 --batch-size 32 --epochs 200 --img 640 --data data/custom_dataset.yaml --cfg models/yolov5s-custom.yaml --weights yolov5s.pt --name yolov5s-custom --resume`

Because when I resume training with the first command, the parameter (batch size, imgsz, etc) for training is in default value, like in this image below

![image](https://github.com/ultralytics/yolov5/assets/88174587/c2ce221d-3b93-4088-90a0-7886f673a1d1)

Before my training is interrupted, the train is showing my pre-defined value (batch size 32, data: custom.yaml)

![image](https://github.com/ultralytics/yolov5/assets/88174587/5bbcd730-e435-4538-aed9-28952244a355)

Can you make it clear that did I have to put all arguments just like before it was? Thank you in advance @glenn-jocher!

**Comment by glenn-jocher** [comment]
@itsjustrafli hello! If you want to resume training with your previously defined parameters, you do need to include them in the command along with the `--resume` argument. This is because `--resume` only resumes the training from the checkpoint file without any additional arguments. 

So to resume training with your previous configurations, you would need to use the following command:

```
!python train.py --device 0 --batch-size 32 --epochs 200 --img 640 --data data/custom_dataset.yaml --cfg models/yolov5s-custom.yaml --weights yolov5s.pt --name yolov5s-custom --resume /path/to/checkpoint_file.pt
```

Please replace `/path/to/checkpoint_file.pt` with the path to your desired checkpoint file. This should allow you to resume training with the correct configurations. Let us know if you have any further questions or concerns!

**Comment by itsjustrafli** [comment]
It's crystal clear now. Once again, thank you so much, @glenn-jocher !

**Comment by glenn-jocher** [comment]
You're welcome, @itsjustrafli! I'm glad I could help you. If you have any further questions or concerns, please feel free to let us know. We're always happy to assist you with your YOLOv5 training needs!

**Comment by RuyangFan** [comment]
If I set epoch=100 for my first training, when I finish training, I want to train 200epoch based on the training results, what should I do

**Comment by glenn-jocher** [comment]
@BigBull-man hi there! If you want to continue training for an additional 100 epochs based on the results of your initial training, you can simply use the `--epochs` argument with a value of 200 when you start the training script again. This will continue training from the last saved checkpoint and extend it for the desired number of epochs. You don't need to change any other settings or parameters. Just make sure to specify the correct path to your existing checkpoint using the `--resume` argument. Let me know if you have any further questions or need more assistance!
